docs/core-concepts/navigation.md

```diff
@@ -0,0 +1,103 @@
+# Navigation
+
+We'll use these example routes to demonstrate navigation:
+
+```ts
+const blog = createRoute({
+  name: 'blog',
+  path: '/blog',
+})
+
+const blogPost = createRoute({
+  parent: blog,
+  name: 'blogPost',
+  path: path('/[blogPostId]', {
+    blogPostId: Number,
+  }),
+})
+```
+
+## Routes vs Urls
+All navigation methods accept a route name or a url. Using a route name is the recommended because it is type safe. But sometimes necessary to use a url. These examples are all the same functionally.
+
+::: code-group
+```ts [Router]
+// type safe
+router.push('blogPost', {
+  blogPostId: 1,
+})
+
+// not type safe
+router.push('/blogPost/1')
+```
+
+```vue [Router Link]
+<!-- type safe -->
+<router-link to="(resolve) => resolve('blogPost', { blogPostId: 1 })">Blog Post One</router-link>
+
+<!-- not type safe -->
+<router-link to="/blogPost/1">Blog Post One</router-link>
+```
+:::
+
+## Push vs Replace
+A push is the default when navigating. This will add a new entry to the browser history using [pushState](https://developer.mozilla.org/en-US/docs/Web/API/History/pushState).
+
+A replace will also navigate to the new route, but it will replace the current entry in the browser history using [replaceState](https://developer.mozilla.org/en-US/docs/Web/API/History/replaceState).
+
+::: code-group
+```ts [Router]
+// push
+router.push('blog')
+
+// replace
+router.replace('blog')
+
+// or
+router.push('blog', { replace: true })
+```
+
+```vue [RouterLink]
+<!-- push -->
+<router-link to="(resolve) => resolve('blog')">Blog</router-link>
+
+<!-- replace -->
+<router-link to="(resolve) => resolve('blog')" replace>Blog</router-link>
+```
+:::
+
+
+## Resolved Routes
+A [ResolvedRoute](/api/types/ResolvedRoute) is the base of what makes up the [Router Route](/core-concepts/router-route). It represents a [route](/core-concepts/routes) that has been matched to a specific location. It includes any params, state, query, and hash values for that location. Resolved routes are how Kitbag Router ensures type safety when navigating. There are a few ways to get a resolved route.
```

Comment:
this last sentence says 
> There are a few ways to get a resolved route.

but then goes on to show 1 way to get a resolved route and 2 other tabs that reference the 1st tab and also somewhat repeat the code-group above

Reply:
I'm struggling to see the value in conveying the "resolved route" concept to users at all

Reply:
Resolved routes are pretty critical to how the router works and how navigation works. So I think looking under the hood a bit here is nice. The idea behind the code group is showing (through repetition) that these all work the same way. Switching between tabs its easy to identify the pieces that are the same and have that "ahh" moment. 

I think its specifically useful considering router link provides a callback called `resolve` and we have other router link examples that show it. Without this section IMO there's a bit of confusion. "what is resolve",  "why do I need to use this function", etc. 

I thought you'd be proud how concise this was 😂 let's discuss